base/interactiveutil.jl

```diff
@@ -254,8 +254,13 @@ function gen_call_with_extracted_types(fcn, ex0)
                     Expr(:call, typesof, map(esc, args[2:end])...))
     end
     exret = Expr(:none)
+    is_macro = false
     ex = expand(ex0)
-    if !isa(ex, Expr)
+    if fcn in [:which, :less, :edit, :functionloc] && ex0.head == :macrocall
```

Comment:
I think you should check `isa(ex0, Expr)` here.

Also, I still don't understand why are the special cases necessary here. Why do you want `@code_lowered @time 1 + 2` to error instead of showing the ast?